compiler+runtime/src/jank/clojure/core.jank

```diff
@@ -6741,20 +6744,23 @@ fails, attempts to require sym's namespace and retries."
   "Opens a reader on f and reads all its contents, returning a string.
   See clojure.java.io/reader for a complete list of supported arguments."
   ([f & opts]
-    ;;  (let [opts (normalize-slurp-opts opts)
-    ;;        sw (java.io.StringWriter.)]
-    ;;    (with-open [^java.io.Reader r (apply jio/reader f opts)]
-    ;;      (jio/copy r sw)
-    ;;      (.toString sw)))
-   (throw "TODO: port slurp")))
+   (let [opts (normalize-slurp-opts opts)
+         file (cpp/std.ifstream. (cpp/cast (cpp/type "char const*") f))
+         buffer (cpp/std.stringstream.)]
+     (when-not (cpp/.is_open file)
+       (throw (str f " (No such file or directory)")))
+     (cpp/<< buffer (cpp/.rdbuf file))
```

Comment:
BTW you can avoid reallocation in the stringstream by measuring the file size ahead of time, allocating a fixed size string to serve as a buffer, and then reading into it. See `load_to_string` in Boost: https://www.boost.org/doc/libs/1_74_0/boost/filesystem/string_file.hpp

Reply:
I tried porting the code from boost via interop and struggled to figure out how to express `&str[0]`. Anything I tried wouldn't match the signature for `ifstream.read`. I could use a _pointer_ if it's easy enough and I'm just missing something. I also pushed up the todo if we want to call this done for now.

Reply:
`&s.front()`/`(cpp/& (cpp/.front s))` is the same thing but doesn't require any array interop.